frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -420,9 +422,25 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
               setBuildingToAssociateLand(building);
               setShowAssociateLandModal(true);
             }}
-            goToAssociatedLand={(building: IBuilding) => {
-              setBuildingToAssociateLand(building);
-              addAssociatedLand();
+            goToAssociatedLand={async (building: IBuilding) => {
+              if (!!formikRef?.current) {
+                const values = formikRef.current.values;
+                const apiValues = valuesToApiFormat(cloneDeep(values));
+                try {
+                  if (!apiValues.id) {
```

Comment:
Is this to save/update the building before going to associated land?

Reply:
when clicking the modify land button, Mikaela wanted the building to automatically save (otherwise the changes to the building wouldn't be saved).